packages/ckeditor5-font/tests/ui/colorui.js

```diff
@@ -194,18 +195,50 @@ describe( 'ColorUI', () => {
 				sinon.assert.notCalled( spy );
 			} );
 
-			it( 'should cancel changes', () => {
-				dropdown.isOpen = false;
-
-				dropdown.colorTableView.selectedColor = '#ff0000';
+			it( 'should undo changes', () => {
+				const spyUndo = sinon.spy( editor.commands.get( 'undo' ), 'execute' );
 
 				dropdown.isOpen = true;
+				testColorPlugin.colorTableView.colorGridsPageView.colorPickerButtonView.fire( 'execute' );
+
+				dropdown.colorTableView.selectedColor = 'hsl( 0, 0%, 100% )';
 
-				dropdown.colorTableView.selectedColor = '#123456';
+				editor.commands.get( 'testColorCommand' ).isEnabled = true;
+
+				dropdown.colorTableView.fire( 'execute', {
+					value: 'hsl( 210, 65%, 20% )',
+					source: 'colorPicker'
+				} );
 
 				dropdown.colorTableView.colorPickerPageView.cancelButtonView.fire( 'execute' );
 
-				expect( dropdown.colorTableView.selectedColor ).to.equal( '#ff0000' );
+				sinon.assert.calledOnce( spyUndo );
+			} );
+
+			it( 'should create new batch when color picker is showed', () => {
```

Comment:
I can't find the test that actually counts the undo steps if there were more than one color change. Could you add it?